File: Arch.Data/Arch.Data/Common/Enums/DatabaseType.cs
using System.ComponentModel;

namespace Arch.Data.Common.Enums
{
    /// <summary>
    /// 物理数据库类型
    /// </summary>
    public enum DatabaseType
    {
        /// <summary>
        /// 主数据库，可读可写
        /// </summary>
        [Description("主数据库")]
        Master = 0,
        /// <summary>
        /// 从数据库，可读
        /// </summary>
        [Description("从数据库")]
        Slave = 1
    }
}

File: dao-gen-core/src/main/java/com/ctrip/platform/dal/daogen/enums/DatabaseType.java
package com.ctrip.platform.dal.daogen.enums;

public enum DatabaseType {
	
	MySQL("Arch.Data.MySqlProvider"),
	SQLServer("System.Sql.SqlClient");
	
	private String value;

	DatabaseType(String value){
		this.value = value;
	}

	public static DatabaseType getInstanceByValue(String value) {
		for (DatabaseType databaseType : DatabaseType.values()) {
			if (databaseType.getValue().equals(value)) {
				return  databaseType;
			}
		}
		return SQLServer;
	}

	public String getValue() {
		return value;
	}

	public void setValue(String value) {
		this.value = value;
	}
}
